src/ucp/api/ucp.h

```diff
@@ -1819,6 +1820,12 @@ typedef struct {
      */
     ucp_mem_h memh;
 
+    /**
+     * Enumerated value, indicates message priority, priority is not guaranteed 
+     * and depends on the capabilities of the selected transport,
+     * ordering across messages with different priorities are also not guaranteed.
```

Comment:
Does it mean that order for messages with the same priorities is guaranteed ? IMHO it becomes really confusing really quickly.

Reply:
how does it co-exist with TAGed message semantics ?

Reply:
> Can you please specify application use case ?
Realm with UCX backend, Realm is required to support priority messages, a kind of data-path / control-path situation in which we don't want the control messages to wait behind large data messages. They currently have a workaround (creating another worker) which shows better results.

> Does it mean that order for messages with the same priorities is guaranteed ? IMHO it becomes really confusing really quickly.
Yes, order is guaranteed as long as you don't play with priorities (which is kind of the point of priority messages)

> how does it co-exist with TAGed message semantics ? we somewhat define ordering semantics on top of other ordering semantics

The main goal is to add priorities to active messages, TAGed messages shouldn't be affected.

Reply:
It is somewhat like adding "tags" to active messages.  In this case it should be explicitly defined as AM message priority.  I would not limit this to low and high. The tag can have a weight.